end2end/tests/reportBuilder.spec.ts

```diff
@@ -2,30 +2,208 @@ import { test, expect } from '@playwright/test';
+    await page.getByRole('button', { name: 'Save Change' }).click();
+    await page.getByRole('cell', { name: 'Available for TEST CHANGE TITLE' }).first().click();
+    await page.getByLabel('Report Title').click();
+    await page.getByLabel('Report Title').fill('Available for test case');
+    await page.getByLabel('Report Title').press('Enter');
+    await page.getByRole('button', { name: 'Save Change' }).click();
+});
+
+test("Connected forms open on UID link click", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHJfNDA0UyhFGhg5dxwGMCRgNRBhNBhIpABGW0LyLJywAHkAMwq4fTsVIA%3D%3D%3D&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgXRiEHeln1Gi6stU8AukA");
+    await page.getByRole('link', { name: '956' }).click();
+    await expect(page.locator('#headerTab')).toContainText('Request #956');
+});
+
+test("Modify Search if updated Generate Report will shows No Result  ", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/");
+    await page.getByText('Report Builder Create custom').click();
+    await page.getByRole('button', { name: 'Next Step' }).click();
+    await page.locator('#indicatorList').getByText('Current Status').click();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+
+    await page.getByRole('button', { name: 'Modify Search' }).click();
+    await page.getByRole('cell', { name: 'Current Status' }).locator('a').click();
+    await page.getByRole('option', { name: 'Initiator' }).click();
+    await page.getByRole('button', { name: 'Next Step' }).click();
+    await page.getByText('General Workflow - Group').click();
+    await page.getByText('General Workflow - Requestor').click();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+    await expect(page.getByRole('cell', { name: 'No Results' })).toBeVisible();
+});
+
+test("Take Action allow user to perform necessary form Action", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('row', { name: '955 Take Action Available for' }).locator('div').click();
+    await page.getByLabel('comment text area').click();
+    await page.getByLabel('comment text area').fill('testing purpose');
+    await page.getByRole('button', { name: 'Close' }).click();
+
```

Comment:
This test is incomplete. The label states it's supposed to take an action, but the procedure closes the modal. Also needs a relevant assertion.

Reply:
resolved